org.eclipse.scout.rt.platform/src/main/java/org/eclipse/scout/rt/platform/opentelemetry/AbstractAsyncObservableMeasurement.java

```diff
@@ -0,0 +1,175 @@
+import java.util.function.Consumer;
+import java.util.function.Supplier;
+
+import org.eclipse.scout.rt.platform.BEANS;
+import org.eclipse.scout.rt.platform.config.AbstractPositiveIntegerConfigProperty;
+import org.eclipse.scout.rt.platform.config.CONFIG;
+import org.eclipse.scout.rt.platform.context.RunContext;
+import org.eclipse.scout.rt.platform.context.RunContexts;
+import org.eclipse.scout.rt.platform.exception.ExceptionHandler;
+import org.eclipse.scout.rt.platform.job.IExecutionSemaphore;
+import org.eclipse.scout.rt.platform.job.IFuture;
+import org.eclipse.scout.rt.platform.job.Jobs;
+import org.eclipse.scout.rt.platform.util.LazyValue;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.opentelemetry.api.metrics.ObservableMeasurement;
+
+/**
+ * {@link ObservableMeasurement} to support "long-running" observable measurements that are independent of
+ * OpenTelemetry's metrics export interval. "long-running" means measurements that are based on values accessing other
+ * (external) systems, e.g. database queries, and can therefore take longer than few milliseconds. To avoid a
+ * time-consuming metrics export process, the measurement is done asynchronously.<br/>
+ * <b>Important:</b> Prefer to use in-memory metrics instead of this type of metrics if possible.
+ * <p>
+ * The observation is executed asynchronously with a predefined minimum interval (see
+ * {@link AsyncObservationIntervalProperty}). The schedule of this async job is controlled during the execution of the
+ * metrics export. As soon as the last job trigger is outside the specified interval, the job is re-triggered during the
+ * metrics export (see {@link #accept(ObservableMeasurement)}).
+ * </p>
+ * <p>
+ * {@link ConcurrentAsyncObservableJobProperty} is used to prevent too many parallel "long-running" observable
+ * measurements, which could cause resource starvation (mainly with database connections).
+ * </p>
+ */
+public abstract class AbstractAsyncObservableMeasurement<M extends ObservableMeasurement, V> implements Consumer<M> {
+
+  private static final Logger LOG = LoggerFactory.getLogger(AbstractAsyncObservableMeasurement.class);
+
+  static final String ASYNC_JOB_NAME_PATTERN = "[otel] Async measurement: {}";
+  static final LazyValue<IExecutionSemaphore> ASYNC_JOB_EXECUTION_SEMAPHORE = new LazyValue<>(() -> Jobs.newExecutionSemaphore(CONFIG.getPropertyValue(ConcurrentAsyncObservableJobProperty.class)));
+  static final String ASYNC_JOB_EXECUTION_HINT = AbstractAsyncObservableMeasurement.class.getSimpleName() + "$ASYNC_JOB";
+
+  private final String m_name;
+  private final Callable<V> m_callable;
+  private final Supplier<RunContext> m_runContextSupplier;
+  private final long m_asyncObservationIntervalInMs;
+
+  private volatile V m_asyncMeasurementValue;
+  private final AtomicReference<IFuture<V>> m_asyncJobRef = new AtomicReference<>();
+  private long m_asyncJobLastTriggerTimestamp;
```

Comment:
If `accept` is always invoked within the same (openTelemetry) thread, it is ok. Otherwise this member should also be marked with `volatile`.

Reply:
In OpenTelemetry the metric collections across all readers are sequential. Therefore, no specific concurrency handling is required.